src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice/GasPriceOracle.cs

```diff
@@ -70,13 +71,39 @@ public UInt256 GetGasPriceEstimate()
             LastGasPrice = gasPriceEstimate;
             return gasPriceEstimate!;
         }
+        
+        public UInt256 GetMaxPriorityGasFeeEstimate()
+        {
+            Block? headBlock = _blockFinder.Head;
+            if (headBlock is null)
+            {
+                return FallbackGasPrice();
+            }
+            
+            if (MaxPriorityFeePerGas is not null && LastHeadBlock!.Hash == headBlock!.Hash)
+            {
+                return MaxPriorityFeePerGas.Value;
+            }
+
+            LastHeadBlock = headBlock;
```

Comment:
If we set LastHeadBlock either from here or GetGasPriceEstimate, wont that break each other logic?

Reply:
Maybe they should be grouped in a struct or tuple to make the pairs differentiate?